test/unit/rules/no-unnecessary-curly-literals-test.js

```diff
@@ -0,0 +1,150 @@
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+generateRuleTests({
+  name: 'no-unnecessary-curly-literals',
+
+  config: true,
+
+  good: [
```

Comment:
We should check component arguments as well, because now the rule also flags stuff like `@foo={{true}}`.

Reply:
Also `attr={{null}}` and `attr=null`, and `attr={{undefined}}` and `attr=undefined` aren't the same thing.
`attr={{null}}` does not render the attribute and `attr=null` renders `attr="null"`.

Reply:
Hmm, was I wrong to suggest this should be expanded to other types of literals? What types of literals should be included in this rule?

Reply:
Hmm I thought that strings should be the only literals applicable to this rule? I believe other literals (like number or boolean) do need to be wrapped in curly braces

Reply:
> Hmm I thought that strings should be the only literals applicable to this rule? I believe other literals (like number or boolean) do need to be wrapped in curly braces

That's what I also think. Or thought. I just tried it with numbers and booleans and... it works. :smile: Never knew that. But I'm not sure it's really recommended by the Ember community/core team. Perhaps it's frowned upon.

Also, note there is the [no-quoteless-attributes rule](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-quoteless-attributes.md).

Reply:
Okay, someone can rename the rule to `no-unnecessary-curly-strings` and we can release it as a bug fix if that's best.

Reply:
Can you folks please review the PR to rename the rule to `no-unnecessary-curly-strings` and confirm that it makes sense now? https://github.com/ember-template-lint/ember-template-lint/pull/2802?